src/shogun/kernel/SpectralMixtureKernel.cpp

```diff
@@ -0,0 +1,221 @@
+}
+
+int32_t CSpectralMixtureKernel::get_num_components() const
+{
+	return m_weights.size();
+}
+
+void CSpectralMixtureKernel::set_component_parameters(const SGVector<float64_t>& weights,
+	const SGVector<float64_t>& means, const SGVector<float64_t>& stddeviations)
+{
+	bool same_size= (weights.size()==means.size()) && (weights.size()==stddeviations.size());
+	REQUIRE(same_size, "Weights, means, and stddeviations vectors must be of equal size.");
+
+	m_weights = weights;
+	m_means = means;
+	m_stddeviations = stddeviations;
+}
+
+float64_t CSpectralMixtureKernel::compute(int32_t idx_a, int32_t idx_b)
+{
+	REQUIRE(has_features(), "Features not set - cannot compute kernel value.");
+	REQUIRE(m_weights.size() > 0, "Mixture components not set - cannot compute kernel values.");
+
+	CDotFeatures* dotlhs = dynamic_cast<CDotFeatures*>(lhs);
+	CDotFeatures* dotrhs = dynamic_cast<CDotFeatures*>(rhs);
+
+	REQUIRE(dotlhs != NULL, "Left-hand-side features must be of type CDotFeatures");
+	REQUIRE(dotrhs != NULL, "Right-hand-side features must be of type CDotFeatures");
+
+	float64_t result = 0.0;
+	float64_t distance_sq = dotlhs->dot(idx_a, dotlhs, idx_a) + dotrhs->dot(idx_b, dotrhs, idx_b) - 2*dotlhs->dot(idx_a, dotrhs, idx_b);
+
+	for(int i=0; i<m_weights.size(); i++)
+		result += CMath::pow(m_weights[i],2)*CMath::exp(-2*CMath::pow(CMath::PI,2)*distance_sq*CMath::pow(m_stddeviations[i],2))*CMath::cos(2*CMath::PI*CMath::sqrt(distance_sq)*m_means[i]);
+
+	return result;
+}
+
+SGVector<float64_t> CSpectralMixtureKernel::get_squared_features(CDotFeatures* features)
+{
+	SGVector<float64_t> sqr_features = SGVector<float64_t>(features->get_num_vectors());
+	
+	for (int i=0; i<sqr_features.size(); i++)
+		sqr_features[i] = features->dot(i, features, i);
+
+	return sqr_features;
+}
+
+SGMatrix<float64_t> CSpectralMixtureKernel::get_square_distance_matrix()
+{
+	CDotFeatures* dot_lhs = dynamic_cast<CDotFeatures*>(lhs);
+	CDotFeatures* dot_rhs = dynamic_cast<CDotFeatures*>(rhs);
+
+	REQUIRE(dot_lhs != NULL, "Left-hand-side features must be of type CDotFeatures");
+	REQUIRE(dot_rhs != NULL, "Right-hand-side features must be of type CDotFeatures");
+	
+	SGMatrix<float64_t> lhs_mtrx = dot_lhs->get_computed_dot_feature_matrix();
+	SGMatrix<float64_t> rhs_mtrx = dot_rhs->get_computed_dot_feature_matrix();
+
+	/* rhs^T . lhs: index (i, j) is dot product of lhs_i and rhs_j.
+	   result scaled by minus two for distance computation */
+	SGMatrix<float64_t> dot_mtrx = SGMatrix<float64_t>::matrix_multiply(rhs_mtrx, lhs_mtrx, true, false, -2.0);
+
+	SGVector<float64_t> sqr_lhs = get_squared_features(dot_lhs);
+	SGVector<float64_t> sqr_rhs = get_squared_features(dot_rhs);
+
+	for (int i=0; i<dot_mtrx.num_cols; i++)
+	{
+		/* To each column vector is added a scalar: lhs_i^2
+		   and the vector [rhs_1^2 ... rhs_n^2]^T*/
+		float64_t* col_vector_ptr = dot_mtrx.get_column_vector(i);
+
+		SGVector<float64_t>::add_scalar(sqr_lhs[i], col_vector_ptr, dot_mtrx.num_rows);
```

Comment:
@karlnapf Add scalar is not yet there in linalg. I will add that there, but I won't be able to do that until Monday (going on a trip).
So I think that we shouldn't keep this PR on hold for that. But @sorig can open as issue regarding replacing it and assign it to me for me to take care of that later?

@sorig you can find most useful linalg methods (very few are there at this moment) in [Core](https://github.com/lambday/shogun/blob/develop/src/shogun/mathematics/linalg/internal/modules/Core.h) and [Redux](https://github.com/lambday/shogun/blob/develop/src/shogun/mathematics/linalg/internal/modules/Redux.h).
These methods take a backend that it will use to perform the computation, which currently are
- NATIVE (our native implementation, this one is available only for a couple of operations as of now), 
- EIGEN3 and 
- VIENNACL. 

I'd suggest that when you use these methods, just 
- include `<shogun/mathematics/linalg/linalg.h>` in your cpp (**NOT** in header), 
- use explicit Eigen3 backend and 
- keep your code `HAVE_EIGEN3` and `HAVE_CXX11 || HAVE_CXX0X` guarded. 

You can check the [README](https://github.com/shogun-toolbox/shogun/wiki/README_linalg) for linalg for some more details.Specially check the part [for Shogun developers](https://github.com/shogun-toolbox/shogun/wiki/README_linalg#for-shogun-developers).